connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -841,6 +848,12 @@ message FetchErrorDetailsResponse {
 
     // The corresponding fragment of the query which throws the exception.
     string fragment = 5;
+
+    // The user code (call site of the API) that caused throwing the exception.
+    optional string callSite = 6;
```

Comment:
Do we need to differentiate null or "" for callSite? If not,
```
string callSite = 6;
```
should be sufficient

Reply:
The actual differentiator is the new field `contextType`, so, we can put a default value.

Reply:
Removed `optional`